scworker.js

```diff
@@ -134,8 +134,9 @@ SCWorker.prototype._init = function (options) {
   }
 
   this.on(this.EVENT_READY, function () {
-    self.start();
-    handleReady();
+    self.start().then(function () {
+      handleReady();
```

Comment:
Also I changed this block of code to only execute `handleReady` if the `start` function properly resolves.